packages/beacon-node/src/network/network.ts

```diff
@@ -243,9 +243,67 @@ export class Network implements INetwork {
   }
 
   async beaconBlocksMaybeBlobsByRoot(peerId: PeerId, request: phase0.BeaconBlocksByRootRequest): Promise<BlockInput[]> {
-    // TODO EIP-4844: Will throw an error for blocks post EIP-4844
-    const blocks = await this.reqResp.beaconBlocksByRoot(peerId, request);
-    return blocks.map((block) => getBlockInput.preEIP4844(this.config, block));
+    // Assume all requests are post EIP-4844
```

Comment:
shouldn't we just do a normal beaconBlocksByRoot request followed by blobs downloading for those blocks which are >= eip4844 epoch?

Reply:
Not sure I follow, why would you query blocks, then blocks+blobs when you can query blocks+blobs directly and save 1 request?

Reply:
`this.config.getForkSeq(this.chain.forkChoice.getFinalizedBlock().slot) >= ForkSeq.eip4844` is not sufficient condition to establish it all post eip4844 blocks, thinking of backfil

Reply:
looks good with the knowledge that BeaconBlocksByRoot is to be primarily used for post finalized

Reply:
Backfill should not use this method. p2p spec says that only unfinalized blocks are server by root. Else use by range. So once the network has finalized the eip4844 no node will serve via p2p a by root request without blobs